### ISSUE TITLE
Test passes when expecting an undefined exception

### ISSUE THREAD

**Issue by fran-f** [pestphp/pest#483]
Hello there 👋 
It seems that Pest 1.21.0 lets a test pass if it throws and expects an exception that *does not exist*.
I can reproduce the issue with the examples below, on both Windows/PHP 7.4.23 and Ubuntu/PHP 7.4.3.

`ProblemTest.php`
```php
<?php
it("throws the exception", function() {
    expect(function() { throw new DoesNotExistException; })
        ->toThrow(DoesNotExistException::class);
});
```

`ControlTest.php`
```php
<?php
class ControlTest extends PHPUnit\Framework\TestCase
{
    public function testExceptionDoesNotExist()
    {
        $this->expectException(DoesNotExistException::class);
        throw new DoesNotExistException;
    }
}
```

Output of `vendor\bin\pest`
```
   FAIL  ControlTest
  ⨯ exception does not exist

   PASS  Tests\ProblemTest
  ✓ it throws the exception

  ---

  • ControlTest > exception does not exist
   PHPUnit\Framework\Exception

  Class DoesNotExistException does not exist

  Tests:  1 failed, 1 passed
  Time:   0.03s
```

The Pest test passes, while the "control" PHPUnit test fails. I would expect Pest to behave as PHPUnit 🤔

[edit] Possibly connected to this check?

https://github.com/pestphp/pest/blob/e64b6fe924df553006db2cdeeee7713f5859f7ac/src/Mixins/Expectation.php#L826,L830

**Comment by fabio-ivona** [comment]
Hi @fran-f thanks for pointing it out!

we are working on a fix for this

**Comment by fabio-ivona** [comment]
@fran-f  the issue is fixed by #487

**Comment by fran-f** [comment]
Thank you @fabio-ivona !
